nkms/data_sources.py

```diff
@@ -1,10 +1,14 @@
 from nkms.crypto.api import encrypt_and_sign
-
+from nkms.crypto.signature import SignatureStamp
+from nkms.keystore.keypairs import SigningKeypair
+from constant_sorrow.constants import NO_SIGNING_POWER
 
 class DataSource:
 
-    def __init__(self, policy_pubkey_enc, signer):
+    def __init__(self, policy_pubkey_enc, signer=NO_SIGNING_POWER):
         self._policy_pubkey_enc = policy_pubkey_enc
+        if signer is NO_SIGNING_POWER:
+            signer = SignatureStamp(SigningKeypair())  # TODO: Generate signing key properly.  #241
```

Comment:
I don't think I like generating the signing key here. I feel like maybe an error should be raised like `NoSigningKey`.

Key generation shouldn't really be an implicit experience, imho. You either have a key available to perform this action, or you don't.

Reply:
Let's talk about this on #241.